plugins/systemair/__init__.py

```diff
@@ -53,6 +55,7 @@ def init_serial_connection(self, serialport, slave_address):
             return False
 
     def _read_modbus(self):
+        self._lockmb.acquire()
```

Comment:
lock release is called from final statement, so it will execute whether an exception has occurred or not (see line 247)